okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http/HttpHeaders.kt

```diff
@@ -225,7 +225,7 @@ fun Response.promisesBody(): Boolean {
   }
 
   val responseCode = code
-  if ((responseCode < HTTP_CONTINUE || responseCode >= 200) &&
+  if ((responseCode < HTTP_CONTINUE || responseCode == HTTP_SWITCHING_PROTOCOLS || responseCode >= 200) &&
```

Comment:
I don’t think I agree that a protocol change is a body? This is used to drive stuff like the gzip interceptor, which seems inappropriate here?

Reply:
The protocol change itself doesn't promise a body, but would depend on the actual protocl _after_ a successful upgrade.
I think this change was some kind of workaround to run into the "newUnknownLengthSource(response.request.url)" case at https://github.com/gesellix/okhttp/blob/0984bc3fc5d41c4814b953194d10e5ee77361f7d/okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http1/Http1ExchangeCodec.kt#L145
Maybe there's a better way?